python/run_benchmark.sh

```diff
@@ -10,115 +10,273 @@
 #     random_forest_classifier
 #     random_forest_regressor
 #
+#     and any comma separted list of the above like knn,linear_regression
+#
+# gpu_etl is gpu ML with Spark RAPIDS plugin for gpu accelerated data loading
+# 
 # By default, the tests should target a single node with a single GPU for CI/CD purposes.
 # For more advanced configurations, use extra <args>.
+# if multiple gpus are available, set CUDA_VISIBLE_DEVICES to comma separated list of gpu indices to use
+#
+# The following environment variables can be set on the command-line to control behavior with the indicated
+# defaults:
+# cuda_version=${cuda_version:-11}
+# cluster_type=${1:-gpu}
+# local_threads=${local_threads:-4}
+# num_rows=${num_rows:-5000}
+# num_cols=${num_cols:-3000}
+# rapids_jar=${rapids_jar:-rapids-4-spark_2.12-$SPARK_RAPIDS_VERSION.jar}
+#
+# ex:  num_rows=1000000 num_cols=300 ./run_benchmark.sh gpu_etl kmeans,pca 
+# would run gpu based kmeans and pca on respective synthetic datasets with 1m rows and 300 cols
+# and would enable the Spark RAPIDS plugin for gpu accelerated data loading.
+
+export CUDA_VISIBLE_DEVICES=${CUDA_VISIBLE_DEVICES:-0}
+cuda_version=${cuda_version:-11}
+
+cluster_type=${1:-gpu}
+shift
+local_threads=${local_threads:-4}
+num_gpus=1
+if [[ $cluster_type == "gpu" || $cluster_type == "gpu_etl" ]]; then
+    num_cpus=0
+    if [ -n $CUDA_VISIBLE_DEVICES ]; then
+        num_gpus=$(( `echo $CUDA_VISIBLE_DEVICES | grep -o ',' | wc -l` + 1 ))
+    fi
+elif [[ $cluster_type == "cpu" ]]; then
+    num_cpus=$local_threads
+    num_gpus=0
+else
+    echo "unknown cluster type $cluster_type"
+    echo "usage: $0 cpu|gpu|gpu_etl mode [extra-args] "
+    exit 1
+fi
+
+num_runs=1
+
 MODE=${1:-all}
 shift
 EXTRA_ARGS=$@
 
 unset SPARK_HOME
-export CUDA_VISIBLE_DEVICES=0
+
+# data set params
+num_rows=${num_rows:-5000}
+knn_num_rows=$num_rows
+num_cols=${num_cols:-3000}
+
+# for large num_rows (e.g. > 100k), set below to ./benchmark/gen_data_distributed.py and /tmp/distributed
+gen_data_script=./benchmark/gen_data.py
+#gen_data_script=./benchmark/gen_data_distributed.py
+gen_data_root=/tmp/data
+
+# if num_rows=1m => output_files=50, scale linearly
+output_num_files=$(( ( $num_rows * $num_cols + 3000 * 20000 - 1 ) / ( 3000 * 20000 ) ))
+
+# if num_cols=3000 => arrow_batch_size=20000, scale linearly for smaller number of columns
+arrow_batch_size=$(( 20000 * ( ( $num_cols + 3000 - 1 ) / $num_cols ) ))
+
 
 # stop on first fail
 set -e
 
+sep="=================="
+
+# TODO adjust maxRecorsPerBatch based on num columns
```

Comment:
Is this comment still relevant?  (If so, there's a typo).

Reply:
No longer relevant.  Deleted.